qoqo_myqlm/interface/myqlm_interface.py

```diff
@@ -128,15 +127,15 @@ def myqlm_call_operation(operation: Any, qureg: qlm.Program.qalloc) -> List:  #
         op = [qlm.ISWAP, qureg[operation.control()], qureg[operation.target()]]
     elif "SingleQubitGate" in tags:
         matrix = operation.unitary_matrix()
-        gate = qlm.AbstractGate("Gate", [], arity=1, matrix_generator=matrix)
+        gate = qlm.AbstractGate(tags[-1], [], arity=1, matrix_generator=lambda: matrix)
```

Comment:
Do the corresponding unittests need adapting?

Reply:
unittest seems to pass with no issues. I believe that the tests were not testing this particular cases, which is about when a gate is outside the one present in qlm and needs to be passed as an abstract one (e.g.VariableMSXX) , with the matrix taken from the operation.unitary_matrix(). I will extend the unittest to reflect those remote cases, but I believe the main code would work already, as I tested manually in a couple of examples